### ISSUE TITLE
chromium uses 100% cpu in dev container and tests timeout

### ISSUE THREAD

**Issue by valankar** [zauberzeug/nicegui#3781]
### Description

I'm trying out the dev container and just running a test (e.g. test_tabs.py). From what I can tell, the chromium process just go to 100% cpu, and after some minutes the test times out.

top output during test run:

```
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
 750002 valankar  20   0 1156.2g  79100  62088 R 100.0   1.0   3:52.08 chromium
 749803 valankar  20   0   32.6g 176452 149428 S   9.3   2.2   0:21.49 chromium
 750067 valankar  20   0   32.6g  46400  31736 S   9.3   0.6   0:20.86 chromium
```

Any idea what is wrong here or how to debug?

**Comment by valankar** [comment]
What vscode looks like during test run:

![image](https://github.com/user-attachments/assets/3be7acf1-829c-4731-953f-90704e85057b)

**Comment by valankar** [comment]
And after the timeout:

![image](https://github.com/user-attachments/assets/ff7a1910-c1ca-418a-9c86-c29064903a5c)

**Comment by rodja** [comment]
This may be a duplicate of https://github.com/zauberzeug/nicegui/issues/3066. Can you check?

**Comment by valankar** [comment]
That looks like something else. In my case it is chromium using up cpu, not a nicegui app. Basically, I'm trying to setup the devcontainer environment to make changes to nicegui code. But when I try any of the nicegui tests they just fail as above.

**Comment by valankar** [comment]
FWIW I'm running Windows 11. I tried both using a WSL2 remote container on local Debian (WSL), as well as a remote SSH on a Debian server. Both result in chromium using 100% cpu and timing out the test. Are you using Debian for the container or something else? I can't tell from the screenshot. Also your process listing shows chrome, whereas mine shows chromium. Not sure if that is relevant.

**Comment by rodja** [comment]
> Are you using Debian for the container or something else?

I have started our devcontainer specified in https://github.com/zauberzeug/nicegui/blob/main/.devcontainer/devcontainer.json which is Debian based. Are you not using this configuration?

**Comment by valankar** [comment]
Yes. I'm referring to the docker host, i.e. the host that creates and runs the devcontainer. At least on Windows, there are a couple of possibilities:

1. Docker daemon running in on a distribution in WSL2
2. Docker daemon running on a remote machine via SSH

Either of these can be any Linux distribution. But it really should not matter, I'm just curious. It sounds like on Mac it is just the docker daemon on Mac.

Is there any Windows developer of NiceGUI that can confirm it works (i.e. not a Mac user)?

I will try to create a video showing the issue.

**Comment by valankar** [comment]
Here is a video showing what I did:

https://youtu.be/rG2LzK5Qf1A?si=YDSZs8ftGLT4kd_C

In this one I'm just using Docker Desktop with WSL2. I cancelled the tests after 10 minutes. The chromium process stays running at 100% cpu and needs to be killed.

**Comment by GrazingScientist** [comment]
I am currently on a Mac using colima and emulating an AMD64 architecture and encountering the same issue.

**Comment by evnchn** [comment]
@valankar @GrazingScientist 

Kindly bump NiceGUI to 2.16.0 and try out `reload=False` `show=False`

Explanation: https://github.com/zauberzeug/nicegui/issues/4521#issuecomment-2845961277

**Comment by falkoschindler** [comment]
@valankar @GrazingScientist
Have you tried upgrading NiceGUI to 2.16+ (or even 3.0+) as @evnchn suggested? Are you still encountering the issue?

**Comment by evnchn** [comment]
#5206 may also help. I have no trust on WSL2 able to run alternative architecture container images.

**Comment by valankar** [comment]
I don't use VScode, dev containers, or Windows anymore.

**Comment by evnchn** [comment]
Good on dropping Windows but I don't see the problem with the first two though. 

If anyone can recreate the issue with a high degree of repeatibility, we can re-open the issue in due course.
